tests/unit_test/action/action_test.py

```
@@ -3303,18 +3309,19 @@ def test_public_search_with_url_exception(self):
             status=500,
             match=[
                 responses.matchers.json_params_matcher({
-                    "text": 'What is AI?', "site": '', "topn": 1
+                    "text": 'What is AI?', "site": '', "topn": 1, "bot": bot
                 })],
         )
 
         with mock.patch.dict(Utility.environment, {'web_search': {"trigger_task": False, "url": search_engine_url}}):
             with pytest.raises(ActionFailure, match=re.escape('Failed to execute the url: Got non-200 status code: 500 {"data": []}')):
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)